com.unity.shadergraph/Editor/Data/Enumerations/Precision.cs

```diff
@@ -1,15 +1,46 @@
+using System;
+
 namespace UnityEditor.ShaderGraph.Internal
 {
+    // this is generally used for user-selectable precision
+    [Serializable]
     enum Precision
     {
-        Inherit,
-        Single,
-        Half,
+        Inherit,    // automatically choose the precision based on the inputs
+        Single,     // force single precision (float)
+        Half,       // force half precision
+        Graph,      // use the graph default (for subgraphs this will properly switch based on the subgraph node setting)
+    }
+
+    // this is used when calculating precision within a graph
+    // it basically represents the precision after applying the automatic inheritance rules
+    // but before applying the fallback to the graph default
+    // tracking this explicitly helps us build subgraph switchable precision behavior (any node using Graph can be switched)
+    public enum GraphPrecision
+    {
+        Single = 0,     // the ordering is different here so we can use the min function to resolve inherit/automatic behavior
+        Graph = 1,
```

Comment:
Looking at some of the code that uses this GraphPrecision enum, I think the naming of the "Graph" value can read as a bit ambiguous, with the "Graph" value that also exists in the Precision enum above

Precision is used for nodes and properties, which can also include SubGraph nodes, and that's fine and makes sense, in that context Graph means use the overall Graph default precision settings, Global Graph Precision if you will

GraphPrecision is used for just SubGraphs, and is represented as "Switchable" in the UI, which conveys its meaning nicely, thats fine too, and it allows for SubGraph nodes to have their precision set when the subgraph node is used in a graph

I guess my concern is just for working with these values (`Precision.Graph` and `GraphPrecision.Graph`) in the code and knowing and understanding the difference between them and how they are meant to be used

If it was perhaps `Precision.Global` and `GraphPrecision.Graph`, or something along those lines, I think it'd be a lot more explicit as to what their respective purposes was.